readme.md
<pre><img width="402" alt="Screenshot 2021-02-21 at 3 51 15 PM" src="https://user-images.githubusercontent.com/64687869/108821634-e716f700-75e3-11eb-8769-9493cde56b83.png"> <img width="401" alt="Screenshot 2021-02-21 at 3 51 58 PM" src="https://user-images.githubusercontent.com/64687869/108821646-ebdbab00-75e3-11eb-87eb-946762adcc26.png">
<img width="416" alt="Screenshot 2021-02-21 at 3 52 10 PM" src="https://user-images.githubusercontent.com/64687869/108821647-ec744180-75e3-11eb-8542-4e43a39ae7f2.png"> <img width="403" alt="Screenshot 2021-02-21 at 3 52 31 PM" src="https://user-images.githubusercontent.com/64687869/108821648-ed0cd800-75e3-11eb-9e48-1d38b152883f.png"></pre>


history.txt
7+1 = 8
675-453+32/12 = 224.66666666666666
785*4536 = 3560760
7*9 = 63
4.2-68 = -63.8
6785-4332 = 2453
784+894 = 1678
7*6 = 42
674+905 = 1579
647-894 = -247
647-894 = -247
647-894 = -247
9/0 = Infinity
5/9 = 0.5555555555555556
4-6 = -2
784+785 = 1569
567/0 = Infinity
764-894 = -130
9) = SyntaxError: Expected token `;'
674-85895*645 = -55401601
674-848 = -174
784*674 = 528416
67*784 = 52528
674*546 = 368004


mainwindow.h
#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include<QString>

QT_BEGIN_NAMESPACE
namespace Ui { class MainWindow; }
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();


private slots:
    void on_pushButton_3_clicked();

    void on_pushButton_2_clicked();

    void on_pushButton_4_clicked();

    void on_pushButton_8_clicked();

    void on_pushButton_7_clicked();

    void on_pushButton_6_clicked();

    void on_pushButton_12_clicked();

    void on_pushButton_11_clicked();

    void on_pushButton_10_clicked();

    void on_pushButton_16_clicked();

    void on_pushButton_15_clicked();

    void on_pushButton_clicked();

    void on_pushButton_5_clicked();

    void on_pushButton_9_clicked();

    void on_pushButton_13_clicked();

    void on_pushButton_14_clicked();

    void on_pushButton_18_clicked();

    void on_pushButton_19_clicked();

    void on_pushButton_17_clicked();

    void on_pushButton_20_clicked();

    void on_lineEdit_returnPressed();

private:
    Ui::MainWindow *ui;

};
#endif // MAINWINDOW_H


mainwindow.cpp
#include "mainwindow.h"
#include "ui_mainwindow.h"
#include<QString>
#include<QJSEngine>
#include<QJSValue>
#include<QFile>
#include<QFileInfo>
#include<QIODevice>
#include<QTextStream>
#include<QMessageBox>


MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::MainWindow)
{
    ui->setupUi(this);
    MainWindow::setFixedSize(this->width(),this->height());
}



MainWindow::~MainWindow()
{
    delete ui;
}



void MainWindow::on_pushButton_3_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"9");
}

void MainWindow::on_pushButton_2_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"8");
}

void MainWindow::on_pushButton_4_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"7");
}

void MainWindow::on_pushButton_8_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"6");
}

void MainWindow::on_pushButton_7_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"5");
}

void MainWindow::on_pushButton_6_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"4");
}

void MainWindow::on_pushButton_12_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"3");
}

void MainWindow::on_pushButton_11_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"2");
}

void MainWindow::on_pushButton_10_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"1");
}

void MainWindow::on_pushButton_16_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"0");
}

void MainWindow::on_pushButton_15_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+".");
}

void MainWindow::on_pushButton_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"+");
}

void MainWindow::on_pushButton_5_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"-");
}

void MainWindow::on_pushButton_9_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"*");
}

void MainWindow::on_pushButton_13_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"/");
}



void MainWindow::on_pushButton_14_clicked()
{
    QJSEngine myEngine;
    QString ques = ui->lineEdit->displayText();
    QJSValue expr = myEngine.evaluate(ques);
    if (expr.isError()){
        ui->lineEdit->setText("Error");
    }
    else{
        ui->lineEdit->setText(expr.toString());

    }

    QString filename = "history.txt";
    QFile file(filename);
    file.open(QIODevice::Append);
    QTextStream stream(&file);
    stream << ques << " = " << expr.toString()<<endl;
    file.close();
}

void MainWindow::on_pushButton_18_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+"(");
}

void MainWindow::on_pushButton_19_clicked()
{
    ui->lineEdit->setText(ui->lineEdit->displayText()+")");
}

void MainWindow::on_pushButton_17_clicked()
{
    ui->lineEdit->setText(QString());
}


void MainWindow::on_pushButton_20_clicked()
{
    QFile file("history.txt");
    file.open(QIODevice::ReadOnly);
    QTextStream stream(&file);
    QString his;
    while(! stream.readLine().isNull()){
        his.append(stream.readLine()+"\n");
    }
    QMessageBox::information(this,QString("History"),his);
}

void MainWindow::on_lineEdit_returnPressed()
{
    MainWindow::on_pushButton_14_clicked();
}


Release/ui_mainwindow.h
/********************************************************************************
** Form generated from reading UI file 'mainwindow.ui'
**
** Created by: Qt User Interface Compiler version 5.12.10
**
** WARNING! All changes made in this file will be lost when recompiling UI file!
********************************************************************************/

#ifndef UI_MAINWINDOW_H
#define UI_MAINWINDOW_H

#include <QtCore/QVariant>
#include <QtGui/QIcon>
#include <QtWidgets/QApplication>
#include <QtWidgets/QHBoxLayout>
#include <QtWidgets/QLineEdit>
#include <QtWidgets/QMainWindow>
#include <QtWidgets/QPushButton>
#include <QtWidgets/QVBoxLayout>
#include <QtWidgets/QWidget>

QT_BEGIN_NAMESPACE

class Ui_MainWindow
{
public:
    QWidget *centralwidget;
    QLineEdit *lineEdit;
    QWidget *layoutWidget;
    QVBoxLayout *verticalLayout;
    QHBoxLayout *horizontalLayout;
    QPushButton *pushButton_20;
    QPushButton *pushButton_19;
    QPushButton *pushButton_18;
    QPushButton *pushButton_17;
    QHBoxLayout *horizontalLayout_2;
    QPushButton *pushButton_13;
    QPushButton *pushButton_3;
    QPushButton *pushButton_2;
    QPushButton *pushButton_4;
    QHBoxLayout *horizontalLayout_3;
    QPushButton *pushButton_9;
    QPushButton *pushButton_8;
    QPushButton *pushButton_7;
    QPushButton *pushButton_6;
    QHBoxLayout *horizontalLayout_4;
    QPushButton *pushButton_5;
    QPushButton *pushButton_10;
    QPushButton *pushButton_11;
    QPushButton *pushButton_12;
    QHBoxLayout *horizontalLayout_5;
    QPushButton *pushButton_16;
    QPushButton *pushButton_15;
    QPushButton *pushButton_14;
    QPushButton *pushButton;

    void setupUi(QMainWindow *MainWindow)
    {
        if (MainWindow->objectName().isEmpty())
            MainWindow->setObjectName(QString::fromUtf8("MainWindow"));
        MainWindow->resize(395, 313);
        QFont font;
        font.setBold(true);
        font.setWeight(75);
        MainWindow->setFont(font);
        QIcon icon;
        icon.addFile(QString::fromUtf8("icon.png"), QSize(), QIcon::Normal, QIcon::Off);
        MainWindow->setWindowIcon(icon);
        MainWindow->setWindowOpacity(0.890000000000000);
        MainWindow->setLayoutDirection(Qt::RightToLeft);
        centralwidget = new QWidget(MainWindow);
        centralwidget->setObjectName(QString::fromUtf8("centralwidget"));
        lineEdit = new QLineEdit(centralwidget);
        lineEdit->setObjectName(QString::fromUtf8("lineEdit"));
        lineEdit->setGeometry(QRect(10, 10, 376, 41));
        QFont font1;
        font1.setPointSize(24);
        font1.setBold(true);
        font1.setWeight(75);
        lineEdit->setFont(font1);
        lineEdit->setAlignment(Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter);
        lineEdit->setReadOnly(false);
        layoutWidget = new QWidget(centralwidget);
        layoutWidget->setObjectName(QString::fromUtf8("layoutWidget"));
        layoutWidget->setGeometry(QRect(10, 60, 371, 241));
        verticalLayout = new QVBoxLayout(layoutWidget);
        verticalLayout->setObjectName(QString::fromUtf8("verticalLayout"));
        verticalLayout->setContentsMargins(0, 0, 0, 0);
        horizontalLayout = new QHBoxLayout();
        horizontalLayout->setObjectName(QString::fromUtf8("horizontalLayout"));
        pushButton_20 = new QPushButton(layoutWidget);
        pushButton_20->setObjectName(QString::fromUtf8("pushButton_20"));

        horizontalLayout->addWidget(pushButton_20);

        pushButton_19 = new QPushButton(layoutWidget);
        pushButton_19->setObjectName(QString::fromUtf8("pushButton_19"));

        horizontalLayout->addWidget(pushButton_19);

        pushButton_18 = new QPushButton(layoutWidget);
        pushButton_18->setObjectName(QString::fromUtf8("pushButton_18"));

        horizontalLayout->addWidget(pushButton_18);

        pushButton_17 = new QPushButton(layoutWidget);
        pushButton_17->setObjectName(QString::fromUtf8("pushButton_17"));

        horizontalLayout->addWidget(pushButton_17);


        verticalLayout->addLayout(horizontalLayout);

        horizontalLayout_2 = new QHBoxLayout();
        horizontalLayout_2->setObjectName(QString::fromUtf8("horizontalLayout_2"));
        pushButton_13 = new QPushButton(layoutWidget);
        pushButton_13->setObjectName(QString::fromUtf8("pushButton_13"));

        horizontalLayout_2->addWidget(pushButton_13);

        pushButton_3 = new QPushButton(layoutWidget);
        pushButton_3->setObjectName(QString::fromUtf8("pushButton_3"));

        horizontalLayout_2->addWidget(pushButton_3);

        pushButton_2 = new QPushButton(layoutWidget);
        pushButton_2->setObjectName(QString::fromUtf8("pushButton_2"));

        horizontalLayout_2->addWidget(pushButton_2);

        pushButton_4 = new QPushButton(layoutWidget);
        pushButton_4->setObjectName(QString::fromUtf8("pushButton_4"));

        horizontalLayout_2->addWidget(pushButton_4);


        verticalLayout->addLayout(horizontalLayout_2);

        horizontalLayout_3 = new QHBoxLayout();
        horizontalLayout_3->setObjectName(QString::fromUtf8("horizontalLayout_3"));
        pushButton_9 = new QPushButton(layoutWidget);
        pushButton_9->setObjectName(QString::fromUtf8("pushButton_9"));

        horizontalLayout_3->addWidget(pushButton_9);

        pushButton_8 = new QPushButton(layoutWidget);
        pushButton_8->setObjectName(QString::fromUtf8("pushButton_8"));

        horizontalLayout_3->addWidget(pushButton_8);

        pushButton_7 = new QPushButton(layoutWidget);
        pushButton_7->setObjectName(QString::fromUtf8("pushButton_7"));

        horizontalLayout_3->addWidget(pushButton_7);

        pushButton_6 = new QPushButton(layoutWidget);
        pushButton_6->setObjectName(QString::fromUtf8("pushButton_6"));

        horizontalLayout_3->addWidget(pushButton_6);


        verticalLayout->addLayout(horizontalLayout_3);

        horizontalLayout_4 = new QHBoxLayout();
        horizontalLayout_4->setObjectName(QString::fromUtf8("horizontalLayout_4"));
        pushButton_5 = new QPushButton(layoutWidget);
        pushButton_5->setObjectName(QString::fromUtf8("pushButton_5"));

        horizontalLayout_4->addWidget(pushButton_5);

        pushButton_10 = new QPushButton(layoutWidget);
        pushButton_10->setObjectName(QString::fromUtf8("pushButton_10"));

        horizontalLayout_4->addWidget(pushButton_10);

        pushButton_11 = new QPushButton(layoutWidget);
        pushButton_11->setObjectName(QString::fromUtf8("pushButton_11"));

        horizontalLayout_4->addWidget(pushButton_11);

        pushButton_12 = new QPushButton(layoutWidget);
        pushButton_12->setObjectName(QString::fromUtf8("pushButton_12"));

        horizontalLayout_4->addWidget(pushButton_12);


        verticalLayout->addLayout(horizontalLayout_4);

        horizontalLayout_5 = new QHBoxLayout();
        horizontalLayout_5->setObjectName(QString::fromUtf8("horizontalLayout_5"));
        pushButton_16 = new QPushButton(layoutWidget);
        pushButton_16->setObjectName(QString::fromUtf8("pushButton_16"));

        horizontalLayout_5->addWidget(pushButton_16);

        pushButton_15 = new QPushButton(layoutWidget);
        pushButton_15->setObjectName(QString::fromUtf8("pushButton_15"));

        horizontalLayout_5->addWidget(pushButton_15);

        pushButton_14 = new QPushButton(layoutWidget);
        pushButton_14->setObjectName(QString::fromUtf8("pushButton_14"));

        horizontalLayout_5->addWidget(pushButton_14);

        pushButton = new QPushButton(layoutWidget);
        pushButton->setObjectName(QString::fromUtf8("pushButton"));

        horizontalLayout_5->addWidget(pushButton);


        verticalLayout->addLayout(horizontalLayout_5);

        MainWindow->setCentralWidget(centralwidget);

        retranslateUi(MainWindow);

        QMetaObject::connectSlotsByName(MainWindow);
    } // setupUi

    void retranslateUi(QMainWindow *MainWindow)
    {
        MainWindow->setWindowTitle(QApplication::translate("MainWindow", "Calculator", nullptr));
        lineEdit->setText(QString());
        pushButton_20->setText(QApplication::translate("MainWindow", "History", nullptr));
        pushButton_19->setText(QApplication::translate("MainWindow", ")", nullptr));
        pushButton_18->setText(QApplication::translate("MainWindow", "(", nullptr));
        pushButton_17->setText(QApplication::translate("MainWindow", "C", nullptr));
        pushButton_13->setText(QApplication::translate("MainWindow", "/", nullptr));
        pushButton_3->setText(QApplication::translate("MainWindow", "9", nullptr));
        pushButton_2->setText(QApplication::translate("MainWindow", "8", nullptr));
        pushButton_4->setText(QApplication::translate("MainWindow", "7", nullptr));
        pushButton_9->setText(QApplication::translate("MainWindow", "*", nullptr));
        pushButton_8->setText(QApplication::translate("MainWindow", "6", nullptr));
        pushButton_7->setText(QApplication::translate("MainWindow", "5", nullptr));
        pushButton_6->setText(QApplication::translate("MainWindow", "4", nullptr));
        pushButton_5->setText(QApplication::translate("MainWindow", "-", nullptr));
        pushButton_10->setText(QApplication::translate("MainWindow", "1", nullptr));
        pushButton_11->setText(QApplication::translate("MainWindow", "2", nullptr));
        pushButton_12->setText(QApplication::translate("MainWindow", "3", nullptr));
        pushButton_16->setText(QApplication::translate("MainWindow", "0", nullptr));
        pushButton_15->setText(QApplication::translate("MainWindow", ".", nullptr));
        pushButton_14->setText(QApplication::translate("MainWindow", "=", nullptr));
        pushButton->setText(QApplication::translate("MainWindow", "+", nullptr));
    } // retranslateUi

};

namespace Ui {
    class MainWindow: public Ui_MainWindow {};
} // namespace Ui

QT_END_NAMESPACE

#endif // UI_MAINWINDOW_H


Release/moc_mainwindow.cpp
/****************************************************************************
** Meta object code from reading C++ file 'mainwindow.h'
**
** Created by: The Qt Meta Object Compiler version 67 (Qt 5.12.10)
**
** WARNING! All changes made in this file will be lost!
*****************************************************************************/

#include "../Calculator/mainwindow.h"
#include <QtCore/qbytearray.h>
#include <QtCore/qmetatype.h>
#if !defined(Q_MOC_OUTPUT_REVISION)
#error "The header file 'mainwindow.h' doesn't include <QObject>."
#elif Q_MOC_OUTPUT_REVISION != 67
#error "This file was generated using the moc from 5.12.10. It"
#error "cannot be used with the include files from this version of Qt."
#error "(The moc has changed too much.)"
#endif

QT_BEGIN_MOC_NAMESPACE
QT_WARNING_PUSH
QT_WARNING_DISABLE_DEPRECATED
struct qt_meta_stringdata_MainWindow_t {
    QByteArrayData data[23];
    char stringdata0[527];
};
#define QT_MOC_LITERAL(idx, ofs, len) \
    Q_STATIC_BYTE_ARRAY_DATA_HEADER_INITIALIZER_WITH_OFFSET(len, \
    qptrdiff(offsetof(qt_meta_stringdata_MainWindow_t, stringdata0) + ofs \
        - idx * sizeof(QByteArrayData)) \
    )
static const qt_meta_stringdata_MainWindow_t qt_meta_stringdata_MainWindow = {
    {
QT_MOC_LITERAL(0, 0, 10), // "MainWindow"
QT_MOC_LITERAL(1, 11, 23), // "on_pushButton_3_clicked"
QT_MOC_LITERAL(2, 35, 0), // ""
QT_MOC_LITERAL(3, 36, 23), // "on_pushButton_2_clicked"
QT_MOC_LITERAL(4, 60, 23), // "on_pushButton_4_clicked"
QT_MOC_LITERAL(5, 84, 23), // "on_pushButton_8_clicked"
QT_MOC_LITERAL(6, 108, 23), // "on_pushButton_7_clicked"
QT_MOC_LITERAL(7, 132, 23), // "on_pushButton_6_clicked"
QT_MOC_LITERAL(8, 156, 24), // "on_pushButton_12_clicked"
QT_MOC_LITERAL(9, 181, 24), // "on_pushButton_11_clicked"
QT_MOC_LITERAL(10, 206, 24), // "on_pushButton_10_clicked"
QT_MOC_LITERAL(11, 231, 24), // "on_pushButton_16_clicked"
QT_MOC_LITERAL(12, 256, 24), // "on_pushButton_15_clicked"
QT_MOC_LITERAL(13, 281, 21), // "on_pushButton_clicked"
QT_MOC_LITERAL(14, 303, 23), // "on_pushButton_5_clicked"
QT_MOC_LITERAL(15, 327, 23), // "on_pushButton_9_clicked"
QT_MOC_LITERAL(16, 351, 24), // "on_pushButton_13_clicked"
QT_MOC_LITERAL(17, 376, 24), // "on_pushButton_14_clicked"
QT_MOC_LITERAL(18, 401, 24), // "on_pushButton_18_clicked"
QT_MOC_LITERAL(19, 426, 24), // "on_pushButton_19_clicked"
QT_MOC_LITERAL(20, 451, 24), // "on_pushButton_17_clicked"
QT_MOC_LITERAL(21, 476, 24), // "on_pushButton_20_clicked"
QT_MOC_LITERAL(22, 501, 25) // "on_lineEdit_returnPressed"

    },
    "MainWindow\0on_pushButton_3_clicked\0\0"
    "on_pushButton_2_clicked\0on_pushButton_4_clicked\0"
    "on_pushButton_8_clicked\0on_pushButton_7_clicked\0"
    "on_pushButton_6_clicked\0"
    "on_pushButton_12_clicked\0"
    "on_pushButton_11_clicked\0"
    "on_pushButton_10_clicked\0"
    "on_pushButton_16_clicked\0"
    "on_pushButton_15_clicked\0on_pushButton_clicked\0"
    "on_pushButton_5_clicked\0on_pushButton_9_clicked\0"
    "on_pushButton_13_clicked\0"
    "on_pushButton_14_clicked\0"
    "on_pushButton_18_clicked\0"
    "on_pushButton_19_clicked\0"
    "on_pushButton_17_clicked\0"
    "on_pushButton_20_clicked\0"
    "on_lineEdit_returnPressed"
};
#undef QT_MOC_LITERAL

static const uint qt_meta_data_MainWindow[] = {

 // content:
       8,       // revision
       0,       // classname
       0,    0, // classinfo
      21,   14, // methods
       0,    0, // properties
       0,    0, // enums/sets
       0,    0, // constructors
       0,       // flags
       0,       // signalCount

 // slots: name, argc, parameters, tag, flags
       1,    0,  119,    2, 0x08 /* Private */,
       3,    0,  120,    2, 0x08 /* Private */,
       4,    0,  121,    2, 0x08 /* Private */,
       5,    0,  122,    2, 0x08 /* Private */,
       6,    0,  123,    2, 0x08 /* Private */,
       7,    0,  124,    2, 0x08 /* Private */,
       8,    0,  125,    2, 0x08 /* Private */,
       9,    0,  126,    2, 0x08 /* Private */,
      10,    0,  127,    2, 0x08 /* Private */,
      11,    0,  128,    2, 0x08 /* Private */,
      12,    0,  129,    2, 0x08 /* Private */,
      13,    0,  130,    2, 0x08 /* Private */,
      14,    0,  131,    2, 0x08 /* Private */,
      15,    0,  132,    2, 0x08 /* Private */,
      16,    0,  133,    2, 0x08 /* Private */,
      17,    0,  134,    2, 0x08 /* Private */,
      18,    0,  135,    2, 0x08 /* Private */,
      19,    0,  136,    2, 0x08 /* Private */,
      20,    0,  137,    2, 0x08 /* Private */,
      21,    0,  138,    2, 0x08 /* Private */,
      22,    0,  139,    2, 0x08 /* Private */,

 // slots: parameters
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,
    QMetaType::Void,

       0        // eod
};

void MainWindow::qt_static_metacall(QObject *_o, QMetaObject::Call _c, int _id, void **_a)
{
    if (_c == QMetaObject::InvokeMetaMethod) {
        auto *_t = static_cast<MainWindow *>(_o);
        Q_UNUSED(_t)
        switch (_id) {
        case 0: _t->on_pushButton_3_clicked(); break;
        case 1: _t->on_pushButton_2_clicked(); break;
        case 2: _t->on_pushButton_4_clicked(); break;
        case 3: _t->on_pushButton_8_clicked(); break;
        case 4: _t->on_pushButton_7_clicked(); break;
        case 5: _t->on_pushButton_6_clicked(); break;
        case 6: _t->on_pushButton_12_clicked(); break;
        case 7: _t->on_pushButton_11_clicked(); break;
        case 8: _t->on_pushButton_10_clicked(); break;
        case 9: _t->on_pushButton_16_clicked(); break;
        case 10: _t->on_pushButton_15_clicked(); break;
        case 11: _t->on_pushButton_clicked(); break;
        case 12: _t->on_pushButton_5_clicked(); break;
        case 13: _t->on_pushButton_9_clicked(); break;
        case 14: _t->on_pushButton_13_clicked(); break;
        case 15: _t->on_pushButton_14_clicked(); break;
        case 16: _t->on_pushButton_18_clicked(); break;
        case 17: _t->on_pushButton_19_clicked(); break;
        case 18: _t->on_pushButton_17_clicked(); break;
        case 19: _t->on_pushButton_20_clicked(); break;
        case 20: _t->on_lineEdit_returnPressed(); break;
        default: ;
        }
    }
    Q_UNUSED(_a);
}

QT_INIT_METAOBJECT const QMetaObject MainWindow::staticMetaObject = { {
    &QMainWindow::staticMetaObject,
    qt_meta_stringdata_MainWindow.data,
    qt_meta_data_MainWindow,
    qt_static_metacall,
    nullptr,
    nullptr
} };


const QMetaObject *MainWindow::metaObject() const
{
    return QObject::d_ptr->metaObject ? QObject::d_ptr->dynamicMetaObject() : &staticMetaObject;
}

void *MainWindow::qt_metacast(const char *_clname)
{
    if (!_clname) return nullptr;
    if (!strcmp(_clname, qt_meta_stringdata_MainWindow.stringdata0))
        return static_cast<void*>(this);
    return QMainWindow::qt_metacast(_clname);
}

int MainWindow::qt_metacall(QMetaObject::Call _c, int _id, void **_a)
{
    _id = QMainWindow::qt_metacall(_c, _id, _a);
    if (_id < 0)
        return _id;
    if (_c == QMetaObject::InvokeMetaMethod) {
        if (_id < 21)
            qt_static_metacall(this, _c, _id, _a);
        _id -= 21;
    } else if (_c == QMetaObject::RegisterMethodArgumentMetaType) {
        if (_id < 21)
            *reinterpret_cast<int*>(_a[0]) = -1;
        _id -= 21;
    }
    return _id;
}
QT_WARNING_POP
QT_END_MOC_NAMESPACE
